monitoring/grafana/dashboards/tests/__init__.py

```diff
@@ -171,7 +171,8 @@ def run_promtool(self):
 
         for variable, value in self.variables.items():
             expr = self.promql_expr_test.expr
-            new_expr = re.sub(r'\${0}'.format(variable), str(value), expr)
+            regex = fr'\${variable}(?=\W)'
```

Comment:
Why the positive look-ahead? What exactly are you trying to match/not to match here? You could express that as a doctest (you'd need to extract part of this function and isolate the string processing code).

Reply:
The positive lookahead is because I don't want to replace that part, but I need it to make a match.

Reply:
@epuertat ping